web/packages/shared/components/MenuLoginWithActionMenu/MenuLoginWithActionMenu.test.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { MenuItem } from 'design';
+import { fireEvent, render, screen, waitFor } from 'design/utils/testing';
+import { MenuInputType } from 'shared/components/MenuLogin';
+
+import { MenuLoginWithActionMenu } from './MenuLoginWithActionMenu';
+
+test('clicking on primary menu and secondary menu opens respective menu items', async () => {
+  render(
+    <MenuLoginWithActionMenu
+      buttonText="Click Me"
+      getLoginItems={() => [{ url: '', login: 'alice' }]}
+      onSelect={() => null}
+    >
+      <MenuItem>Menu item</MenuItem>
+    </MenuLoginWithActionMenu>
+  );
+
+  fireEvent.click(screen.getByText('Click Me'));
+  await waitFor(() => {
+    expect(screen.getByText('alice')).toBeInTheDocument();
+  });
```

Comment:
Every use of `waitFor` followed by `screen.getByText` can be replaced with `await screen.findByText`. `screen.findByText` behaves same as `waitFor` underneath.

```suggestion
  expect(await screen.findByText('alice')).toBeInTheDocument();
```

Reply:
Oh nice I didn't knew it. Thanks!